### ISSUE TITLE
Do not crash with cmake 3.7-3.13 with CMakePresets.json present

### ISSUE THREAD

**Issue by linquize** [microsoft/vscode-cmake-tools#2026]
### Brief Issue Summary

Do not crash with cmake 3.7-3.13 with `CMakePresets.json` present.
If I start VScode directly (cmake auto-detected version is 3.12), or start VSCode in VS2017 command prompt (cmake version is 3.12), this problem exists.
If I start VSCode in Ubuntu 18.04 (cmake from ubuntu bionic is 3.10), this problem exists.

If I start VSCode in VS2019 command prompt (cmake version is 3.20), there is no problem.
If I start VSCode in Ubuntu 20.04 (cmake from focal is 3.16), there is no problem.

Open a workspace with `CMakePresets.json` present.

CMakePresets.json

```json
{
  "version": 2,
  "cmakeMinimumRequired": { "major": 3, "minor": 20, "patch": 0 },
  "configurePresets": [
    {
      "name": "default",
      "generator": "Ninja",
      "binaryDir": "${sourceDir}/build/default"
    }
  ],
  "buildPresets": [ { "name": "default", "configurePreset": "default" }
  ]
}
```

### Expected:

**No Configure Preset Selected** is available for choosing configure preset.

### Apparent Behavior:

Configure Presets list is not available.
If **No Configure Preset Selected** clicked, a toast shows **command 'cmake.selectConfigurePreset' not found**

### CMake Tools Log

```
[variant] Loaded new set of variants
[kit] Successfully loaded 16 kits from C:\Users\User\AppData\Local\CMakeTools\cmake-tools-kits.json
[presetController] Successfully validated presets in c:\project\cmaketest\CMakePresets.json
[rollbar] CMake version 3.12.18081601 may not be supported. Minimum version required is 3.20.0. {}
[main] Unable to determine what CMake generator to use. Please install or configure a preferred generator, or update settings.json, your Kit configuration or PATH variable. Error: No usable generator found.
[expand] Invalid variable reference ${sourceDir} in string: ${sourceDir}/build/default
```

### Platform and Versions

- **Operating System**: Windows
- **CMake Version**: 3.12 (auto detected from VS2017)
- **VSCode Version**: 1.58
- **CMake Tools Extension Version**: 1.7.3
- **Compiler/Toolchain**: Both Visual C++ 2017 and 2019 installed

### Platform and Versions

- **Operating System**: Linux
- **CMake Version**: 3.10 (from Ubuntu bionic)
- **VSCode Version**: 1.58
- **CMake Tools Extension Version**: 1.7.3
- **Compiler/Toolchain**: gcc 7.5, ninja 1.8.2

**Comment by linquize** [comment]
Possible fix:

https://github.com/microsoft/vscode-cmake-tools/blob/e7671157f416c12e85f653042863ffd2c6243993/src/cmake-tools.ts#L1998-L1999

Since the instance is used once only, do not start cmake server if the implementation is `CMakeServerClientDriver` which is serverapi mode that targets cmake 3.7-3.13

**Comment by bobbrow** [comment]
@xisui-MSFT could you take a look at this?

**Comment by xisui-MSFT** [comment]
This issue can be reproed using setting `"cmake.cmakeCommunicationMode": "serverApi"` with later version of CMake. Source dir is required to evaluate presets. Presets contain generator info. Generator info is required for server api. Source dir currently requires driver. So this becomes a circular dependency, and I'll change cmake-tools to use config.sourceDir instead.

**Comment by bobbrow** [comment]
The fix for this was released today in version 1.8.1
